src/client/datascience/notebook/remoteSwitcher.ts

```diff
@@ -0,0 +1,73 @@
+import { DataScience } from '../../common/utils/localize';
+import { noop } from '../../common/utils/misc';
+import { Commands, Settings } from '../constants';
+import { JupyterServerSelector } from '../jupyter/serverSelector';
+import { IJupyterServerUriStorage } from '../types';
+
+@injectable()
+export class RemoteSwitcher implements IExtensionSingleActivationService {
+    private readonly disposables: IDisposable[] = [];
+    constructor(
+        @inject(IJupyterServerUriStorage) private readonly serverUriStorage: IJupyterServerUriStorage,
+        @inject(IDocumentManager) private readonly documentManager: IDocumentManager,
+        @inject(IVSCodeNotebook) private readonly notebook: IVSCodeNotebook,
+        @inject(ICommandManager) private readonly commandManager: ICommandManager,
+        @inject(IDisposableRegistry) private readonly disposableRegistry: IDisposableRegistry,
+        @inject(IApplicationShell) private readonly appShell: IApplicationShell,
+        @inject(JupyterServerSelector) private readonly serverSelector: JupyterServerSelector
+    ) {
+        this.disposableRegistry.push(this);
+    }
+    private statusBarItem!: StatusBarItem;
+    public dispose() {
+        this.disposables.forEach(item => item.dispose());
+    }
+    public async activate(): Promise<void> {
+        this.disposables.push(
+            this.commandManager.registerCommand(
+                Commands.SelectNativeJupyterUriFromStatusBar,
+                this.onStatusBarCommand,
+                this
+            )
+        );
+        this.disposables.push(
+            this.commandManager.registerCommand(Commands.SelectNativeJupyterUriFromToolBar, this.onToolBarCommand, this)
+        );
+        // Set priority of 1000 to ensure it is to the left of the VSC Kernel statusbar item.
+        this.statusBarItem = this.appShell.createStatusBarItem(StatusBarAlignment.Right, 1000);
+        this.notebook.onDidChangeActiveNotebookEditor(this.updateStatusBar.bind(this), this.disposables);
+        this.documentManager.onDidChangeActiveTextEditor(this.updateStatusBar.bind(this), this.disposables);
+        this.serverUriStorage.onDidChangeUri(this.updateStatusBar.bind(this), this.disposables);
+        this.disposables.push(this.statusBarItem);
+        this.updateStatusBar().catch(noop);
+    }
+    private async onStatusBarCommand() {
+        await this.serverSelector.selectJupyterURI(true, 'nativeNotebookStatusBar');
+    }
+    private async onToolBarCommand() {
+        await this.serverSelector.selectJupyterURI(true, 'nativeNotebookToolbar');
+    }
+    private async updateStatusBar() {
+        if (!this.notebook.activeNotebookEditor) {
+            this.statusBarItem.hide();
+            return;
+        }
+        this.statusBarItem.show();
+        const uri = await this.serverUriStorage.getUri();
+        const label =
+            uri === Settings.JupyterServerLocalLaunch
```

Comment:
For remote we're not showing the URI? It's just going to say 'jupyter server: remote'?

Reply:
Yes, else we end up taking a whole lot of space in the status bar.
Ooops, I'm meant to display the Uri in the tooltip.